plugins/vpp/l3plugin/vppcalls/vpp2005/vrrp_dump.go

```diff
@@ -0,0 +1,84 @@
+//  Copyright (c) 2020 Cisco and/or its affiliates.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at:
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+
+package vpp2005
+
+import (
+	"net"
+
+	"go.ligato.io/vpp-agent/v3/plugins/vpp/binapi/vpp2005/ip_types"
+	"go.ligato.io/vpp-agent/v3/plugins/vpp/binapi/vpp2005/vrrp"
+	l3 "go.ligato.io/vpp-agent/v3/proto/ligato/vpp/l3"
+)
+
+// DumpVrrpEntries dumps all configured VRRP entries.
+func (h *VrrpVppHandler) DumpVrrpEntries() (entries []*l3.VRRPEntry, err error) {
+	req := &vrrp.VrrpVrDump{
+		SwIfIndex: 0xffffffff, // Send multirequest to get all VRRP entries
+	}
+	reqCtx := h.callsChannel.SendMultiRequest(req)
+	for {
+		vrrpDetails := &vrrp.VrrpVrDetails{}
+		stop, err := reqCtx.ReceiveReply(vrrpDetails)
+		if stop {
+			break
+		}
+		if err != nil {
+			return nil, err
+		}
+
+		// VRRP interface
+		ifName, _, exists := h.ifIndexes.LookupBySwIfIndex(uint32(vrrpDetails.Config.SwIfIndex))
+		if !exists {
+			h.log.Warnf("VRRP dump: interface name not found for index %d", vrrpDetails.Config.SwIfIndex)
+		}
+
+		var isEnabled bool
+		if vrrpDetails.Runtime.State != vrrp.VRRP_API_VR_STATE_INIT {
+			isEnabled = true
+		}
+
+		isPreempt := (vrrpDetails.Config.Flags & vrrp.VRRP_API_VR_PREEMPT) == vrrp.VRRP_API_VR_PREEMPT
+		isAccept := (vrrpDetails.Config.Flags & vrrp.VRRP_API_VR_ACCEPT) == vrrp.VRRP_API_VR_ACCEPT
+		isUnicast := (vrrpDetails.Config.Flags & vrrp.VRRP_API_VR_UNICAST) == vrrp.VRRP_API_VR_UNICAST
+
+		ipStrs := make([]string, 0, len(vrrpDetails.Addrs))
+		for _, v := range vrrpDetails.Addrs {
+			if v.Af == ip_types.ADDRESS_IP4 {
+				addr := v.Un.GetIP4()
+				ipStrs = append(ipStrs, net.IP(addr[:]).To4().String())
+			} else {
+				addr := v.Un.GetIP6()
+				ipStrs = append(ipStrs, net.IP(addr[:]).To16().String())
+			}
```

Comment:
The [`ip_types.Address`](https://github.com/ligato/vpp-agent/blob/master/plugins/vpp/binapi/vpp2009/ip_types/ip_types.ba.go#L327) used in the [`Addrs`](https://github.com/NotSoFancyName/vpp-agent/blob/vrrp-support/plugins/vpp/binapi/vpp2009/vrrp/vrrp.ba.go#L166) field actually has [`String()`](https://github.com/ligato/vpp-agent/blob/master/plugins/vpp/binapi/vpp2009/ip_types/ip_types.ba.go#L360) method. So you can avoid boilerplate:

```suggestion
			ipStrs = append(ipStrs, v.String())
```

Reply:
My bad, I have not noticed first, it is present in 2009 but not in 2005, so I guess I will leave it this way in 2005 but change in 2009